### PR TITLE
Second draft review

### PR DESCRIPTION
**Pull request by astrofrog**
This PR does not contain a functional branch - it is provided purely for the purpose of providing comments on the first draft of the paper. I have removed style files that don't need to be commented on.

The PDF of this draft can be accessed here: http://mpia.de/~robitaille/astropy/astropy_v3.pdf

### CODE REVIEW

File: astropy.tex [hunk 1]
```diff
@@ -0,0 +1,958 @@
...
+
+\label{sec:acknowledgements}
+
+We would like to thank the \gls{numpy}, \gls{scipy}, \gls{ipython} and
+\gls{matplotlib} communities for providing their packages which are invaluable
+to the development of Astropy. We thank the GitHub team for providing us with
+an excellent free development platform. We also are grateful to Read The Docs
+(\url{https://readthedocs.org/}), Shining Panda
+(\url{https://www.shiningpanda-ci.com/}), and Travis
+(\url{https://www.shiningpanda-ci.com/}) for providing free documentation
```
Comment (hamogu) [inline astropy.tex:951]: If Travis and Shining Panda are refereced with the same url, why not write "Shining Panda and Travis (https//....)"?

File: astropy.tex [hunk 2]
```diff
@@ -0,0 +1,958 @@
...
+coordinate and time transformations, world coordinate system (WCS) support,
+generalized containers for representing gridded as well as tabular data, and a
+framework for cosmological transformations and conversions. Significant
+functionality is under active development, such as a model fitting framework,
+VO client and server tools, and aperture and point spread function (PSF)
+photometry tools. The core development team is actively making additions and
+enhancements to the current code base, and we encourage anyone interested to
+participate in the development of future Astropy versions.
+}
+
```
Comment (hamogu) [inline astropy.tex:230]: Somewhere her should be keywords (although it's not easy to choose the right ones). I suggest:

```
\keywords{Methods: data analysis -- Methods: miscellaneous -- Virtual Observatory Tools}
```

File: astropy.tex [hunk 3]
```diff
@@ -0,0 +1,958 @@
...
+The \texttt{astropy.units} sub-package provides support for physical units. It
+originates from code in the \texttt{pynbody}
+package\footnote{\url{https://code.google.com/p/pynbody/}}, but has been
+significantly enhanced in behavior and implementation. This sub-package can be
+used to attach units to scalars and arrays, convert from one set of units to
+another, define custom units, define equivalencies for units that are not
+strictly the same (such as wavelength and frequency), and decompose units into
+base units. Unit definitions are included in both the International System of
+Units (SI) and the Centimeter-gram-second (CGS) systems, as well as a number of
```
Comment (aconley) [inline astropy.tex:317]: Shouldn't this be Centimeter-Gram-Second?

File: astropy.tex [hunk 4]
```diff
@@ -0,0 +1,958 @@
...
+many methods of the cosmology object as described in the Astropy documentation.
+For users who are more comfortable using a procedural coding style, these
+methods are also available as functions that take a cosmology class instance as
+a keyword argument.
+
+The sub-package provides several pre-defined cosmology instances corresponding
+to commonly used cosmological parameter sets. Currently parameters from the
+WMAP 5-year \citep{Komatsu09}, 7-year \citep{Komatsu11} and 9-year results
+\citep{Hinshaw13} are included (\texttt{WMAP5}, \texttt{WMAP7} and
+\texttt{WMAP9}). There are several classes corresponding to non-flat
```
Comment (aconley) [inline astropy.tex:807]: Should add a mention of the Planck 2013 cosmology (Planck paper XVI: http://adsabs.harvard.edu/abs/2013arXiv1303.5076P)

File: astropy.tex [hunk 5]
```diff
@@ -0,0 +1,958 @@
...
+functionality is under active development, such as a model fitting framework,
+VO client and server tools, and aperture and point spread function (PSF)
+photometry tools. The core development team is actively making additions and
+enhancements to the current code base, and we encourage anyone interested to
+participate in the development of future Astropy versions.
+}
+
+\maketitle
+
+\section{Introduction}
```
Comment (pllim) [inline astropy.tex:233]: This section appears improperly indented at the bottom of Page 1 on the PDF distributed.

**Comment by mservillat** [comment]
Hi all, thanks for putting together this nice paper. I went through the draft and noted a few comments:
- VO is found twice in the abstract, but it is not found in the text except in section 4. I think it would be best to add a paragraph in the introduction, saying what is VO and how it can connect with Astropy.
- section 2.1.1, paragraph 2 : the first sentence is too long and I would remove the end ("such that ...")
- section 2.3 Celestial coordinates: 
  End of paragraph 1: what are the "lessons learned" exactly? Is it what is described in the next paragraph? then a clear link is needed (add "For example", "In particular"...). 
  In general, this section is longer and more difficult to follow. Maybe it could be shortened? maybe some details are not strictly needed, e.g. how transformation graphs work in the last paragraph.
- section 2.6 WCS: 
  naive question, but is there any link (or future link) with the coordinates subpackage? I think some people will have the question in mind at the end of this section.
- section 2.7, paragraph 1: "cosmologies that may be not be flat" --> remove first "be"
- section 3: there is nothing about the license for Astropy (just a word in the intro about some affiliated packages not being compatible with the Astropy license). Shouldn't we say a word about the license?
- section 5, last paragraph: It might be important to give the explicit commands to install astropy (ie, the pip command), to report issues, or to join the development team (maybe in the introduction, or in a specific section). This would give the possibility to many people who won't waste their time going further than the paper to actually become new users and thus get involved.
- Affiliation: being still a Harvard associate, my affiliation would be first AIM (^26) and second CfA (^4)

Cheers,
Mathieu

**Comment by timj** [comment]
I'm sorry I'm not sure how to comment on every little line of the latex in this pull as I've reviewed it the old fashioned way by annotating the PDF.
- Please update my affiliation to also add "Department of Astronomy, Cornell University, Ithaca, NY 14853"
- I'm a bit uncomfortable with so much space in the abstract being devoted to "future work" 
- In the introduction when we refer to the last decade we should include some references. I suggest at least 2011ASPC..442..425G and 2000ASPC..216…59G and maybe a non-STScI reference.
- At the end of the first paragraph where we state that duplication and lack of homogeneity is a problem it would be nice if we could add examples to demonstrate the problem. "What was duplicated?" "citation needed" as wikipedia would say.
- The FITS Standard Group (2008) reference is probably lacking an author name inside the parens. I guess an incorrect \cite variant.
- At the end of section 2.1.1 can we rephrase without the footnote or are we assuming that we will have fixed the problem by the time the paper goes live. Saying "all" and then footnote of "not" is not a good way of putting it.
- The reference to SOFA is incorrect. That reference is the SLALIB reference. For time-based use of SOFA you should reference 2011Metro..48S.200W
- For Celestial coordinates pyast is 2012ASPC..461..825B
- Can we consider adding palpy to the list of pre-existing celestial coordinates packages that we learned from? The reference is still in press but might be out in time

``` bibtex
    @inproceedings{ palpy_adassxii,
       author = {Tim Jenness and David S. Berry},
       booktitle = "ADASS XXII",
       year = 2013,
       editor = {D. Friedel and M. Freemon and R. Plante},
       series = "ASP Conf Ser.",
       volume = "TBD",
       pages = "in press",
       publisher = "ASP",
       address = "San Francisco"
     }
```
- In section 2.3 where we talk about spotting hms in strings to determine that they are hours, could we also have an example that uses the degree symbol. That's a really interesting feature.
- At the top of page 5 we mention IAU 2000/2006 celestial to horizontal coordinates at the end of two consecutive paragraphs. I think one of those needs to be removed or the information merged. We should state whether we support the modern nutation and precession models in general (I don't think we do but ERFA/SOFA does).
- We talk about distances with redshifts. Do we want to mention that we intend to add radial velocities and local standards of rest such as LSRK and heliocentric?
- In section 2.4 maybe cite Pandas rather than use a footnote URL?
- In section 2.5.2 maybe cite a Stetson DAOPHOT ADASS paper? We provide urls to the other formats.
- Section 2.6 we cite an unpublished paper for SIP. We shouldn't do that (it's not different from saying "in prep" so generally shouldn't get a proper reference). SIP seems to be 2005ASPC..347..491S
- The Derriere reference has a problem with its bibtex. Needs curly braces around some of the text.

Otherwise it's looking pretty good. I still worry that A&A will refuse it, especially now that Astronomy & Computing explicitly state that they want software release papers.

**Comment by astrofrog** [comment]
@timj - thanks for the comments! quick question about Pandas - do they have a publication? I always prioritized citing a paper where possible, but I couldn't find one for them.

**Comment by cdeil** [comment]
I think the paper is very nice!
I only have very minor comments which I'll try to leave inline.

File: astropy.tex [hunk 6]
```diff
@@ -0,0 +1,958 @@
...
+    \and
+  Laboratoire AIM, CEA Saclay, Bat. 709, 91191 Gif-sur-Yvette, France
+  \label{inst:saclay}
+    \and
+  Leibniz Institute for Astrophysics Potsdam (AIP)
+  \label{inst:leibniz}
+}
+
+\abstract{
+We present the first public version of the open-source and community-developed
```
Comment (cdeil) [inline astropy.tex:215]: Mention that this paper is for version 0.2 in the abstract?
This would make it easier for the reader to associate each paper with an astropy version in the future.

File: astropy.tex [hunk 7]
```diff
@@ -0,0 +1,958 @@
...
+\vspace{0.1in}
+\fbox{\includegraphics[scale=0.82,clip]{examples/code_constants.pdf}}
+\end{figure}
+
+The \texttt{Quantity} objects are used to define a number of useful
+astronomical constants included with \astropy, each with an associated unit
+(where applicable) and additional metadata describing their provenance and
+uncertainties. These can be used along with \texttt{Quantity} objects to
+provide a convenient framework for computing any quantity in astronomy.
+Figure~\ref{code:constants} includes a simple example that shows how the
```
Comment (cdeil) [inline astropy.tex:396]: Mention the sub-package name `astropy.constants` somewhere in this paragraph?

File: astropy.tex [hunk 8]
```diff
@@ -0,0 +1,958 @@
...
+
+The most common way to use \texttt{astropy.time} is to create a \texttt{Time}
+object by supplying one or more input time values as well as the time
+format or representation and time scale of those values. The input time(s) can
+either be a single scalar like \verb|"2010-01-01 00:00:00"| or \verb|2455348.5|
+or a sequence of such values; the format or representation specifies how to
+interpret the input values, such as ISO or JD or Unix time; and the scale
+specifies the time standard used for the values, such as Coordinated Universal
+Time (UTC), Terrestial Time (TT), or International Atomic Time (TAI). The full
+list of available time scales are listed in Table~\ref{tab:time_systems}. Many
```
Comment (cdeil) [inline astropy.tex:420]: Maybe rephrase "full list ... are listed" to "full list ... is given" or "full list ... is shown" to avoid the duplication?

File: astropy.tex [hunk 9]
```diff
@@ -0,0 +1,958 @@
...
+
+\label{sec:time}
+
+% Authors: Tom A.
+
+The \texttt{astropy.time} package provides functionality for manipulating times
+and dates. Specific emphasis is placed on supporting time scales (e.g. UTC,
+TAI, UT1) and time formats/representations (e.g. JD, MJD, ISO 8601) that are
+used in astronomy. Examples of using this sub-package are provided in Figure
+\ref{code:time}.
```
Comment (cdeil) [inline astropy.tex:410]: E.g. for coordinates there were quite a few citations.
Is there a good review article on time scales and formats?
If so citing it here could be helpful, I think.

File: astropy.tex [hunk 10]
```diff
@@ -0,0 +1,958 @@
...
+demonstrated in Figure~\ref{code:quantities}.
+
+Using the \texttt{.to()} method, \texttt{Quantity} objects can easily be
+converted to different units. The units must either be dimensionally
+equivalent, or users should pass equivalencies through the
+\texttt{equivalencies} argument (c.f. \S\ref{sec:units}).
+
+\begin{figure}
+\center
+\caption{Using the \texttt{astropy.constants} sub-package.\label{code:constants}}
```
Comment (cdeil) [inline astropy.tex:386]: Add `from astropy import units as u` here so that this code example works standalone like all other code examples.

File: astropy.tex [hunk 11]
```diff
@@ -0,0 +1,958 @@
...
+floating-point values, which enables extremely high precision time
+computations. Using two 64-bit floating-point values allows users to represent
+times with a dynamic range of 30 orders of magnitude, for instance times
+accurate to better than a nanosecond over timescales of tens of Gyr. All time
+scale conversions are done by vectorized versions of the SOFA routines using
+Cython\footnote{\url{http://www.cython.org/}} \citep{cython}, a Python package
+that makes it easy to use C code in Python.
+
+\begin{table}
+\caption{Supported time systems for \texttt{astropy.time}\label{tab:time_systems}}
```
Comment (cdeil) [inline astropy.tex:443]: "time systems" -> "time scales"?
(the term "time scale" is used everywhere else and "time system" not mentioned somewhere else I think).

File: astropy.tex [hunk 12]
```diff
@@ -0,0 +1,958 @@
...
+examples for \texttt{astropy.coordinates}. Coordinate objects are created using
+standard Python object instantiation via a Python class named after the
+coordinate system (e.g., \texttt{ICRSCoordinates}). Astronomical coordinates
+may be expressed in a myriad of ways: the classes support string, numeric, and
+tuple value specification through a sophisticated input parser. A design goal
+of the input parser is to be able to determine the angle value and unit from
+the input alone if a person can unambiguously determine them. For example, an
+astronomer seeing the input string ``12h53m11.5123s'' would understand the
+units to be in hours, minutes, and seconds, so this value is alone sufficient
+to pass to the angle initializer. This functionality is build around the
```
Comment (cdeil) [inline astropy.tex:501]: "is build" -> "is built"? (I'm not sure)

File: astropy.tex [hunk 13]
```diff
@@ -0,0 +1,958 @@
...
+ecliptic systems, supergalactic coordinates, and all necessary intermediate
+coordinate systems for the IAU 2000/2006 equatorial-to-horizontal mapping
+\citep[e.g.,][]{soffel03, usnocircular179}.
+
+Another major feature of \texttt{astropy.coordinates} is the methodology for
+transforming between coordinate systems. Figure \ref{fig:code_coordinates}
+illustrates the most basic use of this functionality to convert from ICRS to
+Galactic coordinates. Transformations are provided between all coordinate
+systems built into version 0.2 of Astropy, with the exception of conversions
+from celestial to horizontal coordinates. This is planned for the next major
```
Comment (cdeil) [inline astropy.tex:531]: The last sentence of this paragraph has almost the same content (and gives the same references) as the last sentence of the previous paragraph. Avoid the duplication?

File: astropy.tex [hunk 14]
```diff
@@ -0,0 +1,958 @@
...
+extensible, making it easy for users to define their own readers and writers for
+any other ASCII formats.
+
+\subsubsection{Virtual Observatory tables}
+
+\label{sec:votable}
+
+% Authors: Mike D.
+
+The \texttt{astropy.io.votable} package (formerly the standalone project
```
Comment (cdeil) [inline astropy.tex:717]: "package" -> "sub-package"?
(doesn't matter much, but the term "sub-package" is used consistently throughout the rest of the paper.)

File: astropy.tex [hunk 15]
```diff
@@ -0,0 +1,958 @@
...
+\subsection{Cosmology}
+
+\label{sec:cosmology}
+
+% Authors: Neil Crighton, Alex Conley
+
+The \texttt{astropy.cosmology} sub-package contains classes for representing
+widely used cosmologies, and functions for calculating quantities that depend
+on a cosmological model. It also contains a framework for working with less
+frequently employed cosmologies that may be not be flat, or have a time-varying
```
Comment (cdeil) [inline astropy.tex:773]: "may be not be" -> "may not be"

File: astropy.tex [hunk 16]
```diff
@@ -0,0 +1,958 @@
...
+represented by a class. An instance of this class has attributes giving all the
+parameters required to uniquely specify the cosmology, such as the Hubble
+parameter, CMB temperature and the baryonic, cold dark matter, and dark energy
+densities at $z=0$. One can then use methods of this class to perform
+calculations using these parameters.
+
+Figure~\ref{code:cosmology} shows how \texttt{FlatLambdaCDM} can be used to
+create an object representing a flat $\Lambda$CDM cosmology, and how the
+methods of this object can be called to calculate the comoving volume, age and
+transverse at a given redshift. Further calculations can be performed using the
```
Comment (cdeil) [inline astropy.tex:797]: "transverse at" -> "transverse separation at"

File: astropy.tex [hunk 17]
```diff
@@ -0,0 +1,958 @@
...
+\label{sec:acknowledgements}
+
+We would like to thank the \gls{numpy}, \gls{scipy}, \gls{ipython} and
+\gls{matplotlib} communities for providing their packages which are invaluable
+to the development of Astropy. We thank the GitHub team for providing us with
+an excellent free development platform. We also are grateful to Read The Docs
+(\url{https://readthedocs.org/}), Shining Panda
+(\url{https://www.shiningpanda-ci.com/}), and Travis
+(\url{https://www.shiningpanda-ci.com/}) for providing free documentation
+hosting an testing respectively. Finally, we would like to thank all the
```
Comment (cdeil) [inline astropy.tex:952]: "an" -> "and"

File: astropy.tex [hunk 18]
```diff
@@ -0,0 +1,958 @@
...
+}
+
+\maketitle
+
+\section{Introduction}
+
+% Authors: Tom R., Erik T., Perry G.
+
+The Python programming language\footnote{\url{http://www.python.org}} has been
+one of the fastest-growing programming language in the astronomy community in
```
Comment (cdeil) [inline astropy.tex:238]: "language" -> "languages"?

File: astropy.tex [hunk 19]
```diff
@@ -0,0 +1,958 @@
...
+astronomer seeing the input string ``12h53m11.5123s'' would understand the
+units to be in hours, minutes, and seconds, so this value is alone sufficient
+to pass to the angle initializer. This functionality is build around the
+\texttt{Angle} object, which can be instantiated and used on its own. It
+provides additional functionality like string formatting and mechanisms to
+specify the valid bounds of an angle.
+
+\begin{figure}
+\center
+\caption{Celestial coordinate representation and conversion.\label{code:coords}}
```
Comment (cdeil) [inline astropy.tex:508]: Briefly mention the SIMBAD query feature shown in this Figure in the main text?

**Comment by cdeil** [comment]
If pandas doesn't have a paper we could maybe cite Wes's book?
http://shop.oreilly.com/product/0636920023784.do
It's a good reference for astronomers to get started with Python, I think, so mentioning it in the astropy paper would be nice, I think.

**Comment by cdeil** [comment]
That's all from my side.

Did someone check that all code examples work as advertised (with 0.2 and hopefully also with current master), or would it be helpful if I write them up as `test_v0.2_paper_example.py` unit tests?

**Comment by astrofrog** [comment]
@cdeil - I don't think one person has tested all the examples. I agree it would be good if as well as working with 0.2, they were compatible with the latest dev version. If you are interested in writing these up as a single test file, please feel free to!

**Comment by phn** [comment]
Will it be useful to provide the url http://www.astropy.org/ in the abstract?

File: astropy.tex [hunk 20]
```diff
@@ -0,0 +1,958 @@
...
+TT     & Terrestrial Time \\
+UT1    & Universal Time \\
+UTC    & Coordinated Universal Time \\
+\hline
+\end{tabular}
+\end{table}
+
+\begin{figure}
+\center
+\caption{Time representation and conversion using the \texttt{astropy.time}
```
Comment (cdeil) [inline astropy.tex:462]: In this code example add `...`, i.e. change

```
>>> t = Time(’2010-06-01 00:00:00’,
             format=’iso’, scale=’utc’)
```

to 

```
>>> t = Time(’2010-06-01 00:00:00’,
...          format=’iso’, scale=’utc’)
```

**Comment by cdeil** [comment]
I'm writing up the examples as a test file now.
One thing I noticed is that examples that contain a string can't be copy and pasted from the PDF because the single-quote character is represented as a non-ascii character, resulting in a SyntaxError:

```
In [7]: import astropy.coordinates as coords

In [8]: c = coords.ICRSCoordinates(’00h42m44.3s +41d16m9s’)
  File "<ipython-input-8-586e80f2826a>", line 1
    c = coords.ICRSCoordinates(’00h42m44.3s +41d16m9s’)
                               ^
SyntaxError: invalid syntax
```

This is not very important, but if you can easily make this work via some latex option or by replacing the single with a double quote, it might be worth it.

**Comment by cdeil** [comment]
For the cosmology example (Figure 7) I get a completely different result with astropy master:

```
In [9]: from astropy.cosmology import FlatLambdaCDM

In [10]: cosmo = FlatLambdaCDM(H0=70, Om0=0.3)

In [11]: cosmo.comoving_volume(6.5)
Out[11]: 2521696198211.6924
```
- astropy master: 2521696198211.6924
- current paper draft: 1074707289417.6837

Which result is right?

**Comment by cdeil** [comment]
Here's the unit test for the examples:
https://gist.github.com/cdeil/5727686

The ones for Figures 5 and 6 are not finished ... the example data files need to be generated.
I don't have time now, but I could finish this tonight and make a pull request.

Tom, so you want this test file in the astropy-v0.2-paper repo or somewhere else?

**Comment by cdeil** [comment]
Any chance to get the astropy logo on the front page of the paper?
It's just so nice and the paper doesn't have any images ... :-)

astropy_banner_96

**Comment by astrofrog** [comment]
@cdeil - thanks for doing this! if you could make a pull request to put it in this repository, that would be great!

**Comment by astrofrog** [comment]
@cdeil @nhmc - I get the same result for 0.2.x and the latest developer version, I think the example must have changed but not the result, so it's probably just a typo - @nhmc can you confirm?

**Comment by cdeil** [comment]
I should mention that I didn't check any of the unit conversion and cosmology examples against e.g. result from online calculators. All the unit tests do at the moment is assure astropy gives the output shown in the paper and if tested against master prevent accidental regressions in the future.

**Comment by nhmc** [comment]
The comoving volume error must be typo. 1074707289417.6837 is the volume to z=3, so I must have made a mistake when I copied the example into the draft. The correct value is 2521696198211.6924. Should I open a PR?

**Comment by astrofrog** [comment]
@nhmc - thanks for tracking this down! I can just fix it when I address the other comments.

**Comment by nhmc** [comment]
Ok, great. Thanks for the comments @cdeil!

File: astropy.tex [hunk 21]
```diff
@@ -0,0 +1,958 @@
...
+various reasons: for some the functionality is in early stages of development
+and is not robust; the license is not compatible with Astropy; the package
+includes large files; or the functionality is mature, but too domain-specific
+to be included in a core package.
+
+The driving interface design philosophy behind the core package is that code
+using \astropy should result in concise and easily readable code, even by those new to
+Python. Typical operations should appear in code similar to how they would
+appear if expressed in spoken or written language. Such an interface results in
+code that is less likely to contain errors and is easily readable, enabling
```
Comment (nhmc) [inline astropy.tex:276]: 'and is easily readable' -> 'and is easily understood'

**Comment by timj** [comment]
@astrofrog I was thinking of his O'Reilly book for Pandas.

**Comment by weaverba137** [comment]
In the affiliations, line 18:

'New York, University' -> 'New York University'

Otherwise, I think it looks good!

File: astropy.tex (hunk 9 above)
Comment (timj) [inline astropy.tex:410]: Pat Wallace's Metrologia article that I mentioned in my comments has some of this covered.  You could also cite 1988A&A...194..304G and 2001jsrs.meet..218K 

2006IAUJD..16E..21P seems plausible from the abstract but there is no link on ADS to text.

File: astropy.tex [hunk 22]
```diff
@@ -0,0 +1,958 @@
...
+This sub-package includes the concept of ``equivalencies'' in units, which is
+intended to be used where there exists an equation that provides a relationship
+between two different physical quantities such that providing either one
+uniquely defines the other. A standard astronomical example is the
+relationships between the frequency, wavelength and energy of a photon - it is
+common practice to treat such units as equivalent even though they are not
+strictly comparable. This sub-package supports defining such relationships, and
+it is one of the areas in which it distinguishes itself from most other unit
+handling software. For example, wavelength and frequency are not normally
+convertible, but by passing an equivalency list, the conversion is supported
```
Comment (nhmc) [inline astropy.tex:347]: 'This sub-package supports defining such relationships, and it is one of the areas in which it distinguishes itself from most other unit handling software.'

->

'The inclusion of these equivalencies is an important improvement over existing unit-handling software, which typically does not have this functionality.'

File: astropy.tex (hunk 1 above)
Comment (astrofrog) [inline astropy.tex:951]: This was a mistake - fixed.

File: astropy.tex (hunk 2 above)
Comment (astrofrog) [inline astropy.tex:230]: Good idea - thanks.

File: astropy.tex (hunk 3 above)
Comment (astrofrog) [inline astropy.tex:317]: Fixed

File: astropy.tex (hunk 4 above)
Comment (astrofrog) [inline astropy.tex:807]: This cosmology is not in 0.2 so not in the scope of this paper, but I'll add a mention of it and say it will be in the next release.

File: astropy.tex (hunk 5 above)
Comment (astrofrog) [inline astropy.tex:233]: There's not much I can do about that for now, but I think the editorial office will fix it.

**Comment by DougBurke** [comment]
In section 2.2 I see a double quote mark just underneath the 2 in 2010-01-01... and for the o in the word or that follows this. I see something similar in fig 4 for the label M32.

Is this just the PDF viewer on my tablet messing with me?

edit: for future reference it was my tablet's PDF viewer, since it looks okay on my raspberry pi (high teach review going on here)

File: astropy.tex (hunk 6 above)
Comment (astrofrog) [inline astropy.tex:215]: Fixed

File: astropy.tex (hunk 7 above)
Comment (astrofrog) [inline astropy.tex:396]: Fixed

File: astropy.tex (hunk 8 above)
Comment (astrofrog) [inline astropy.tex:420]: Fixed

File: astropy.tex (hunk 10 above)
Comment (astrofrog) [inline astropy.tex:386]: Fixed

File: astropy.tex (hunk 11 above)
Comment (astrofrog) [inline astropy.tex:443]: Fixed

File: astropy.tex (hunk 12 above)
Comment (astrofrog) [inline astropy.tex:501]: Fixed

File: astropy.tex (hunk 14 above)
Comment (astrofrog) [inline astropy.tex:717]: Fixed

File: astropy.tex (hunk 15 above)
Comment (astrofrog) [inline astropy.tex:773]: Fixed

File: astropy.tex (hunk 16 above)
Comment (astrofrog) [inline astropy.tex:797]: Fixed

File: astropy.tex (hunk 17 above)
Comment (astrofrog) [inline astropy.tex:952]: Fixed

File: astropy.tex (hunk 18 above)
Comment (astrofrog) [inline astropy.tex:238]: Fixed

File: astropy.tex (hunk 20 above)
Comment (astrofrog) [inline astropy.tex:462]: Fixed

File: astropy.tex (hunk 22 above)
Comment (astrofrog) [inline astropy.tex:347]: Done - thanks!

**Comment by astrofrog** [comment]
@DougBurke - I don't see that, can you send me a screenshot?

**Comment by astrofrog** [comment]
@cdeil - the examples can be copied and pasted if I switch to double quotes, so I've done that - thanks for the suggestion!

File: astropy.tex [hunk 23]
```diff
@@ -0,0 +1,958 @@
...
+  \and
+Victor Zabalza\inst{\ref{inst:mpik}}  % confirmed
+  \and
+Zachary Edwards\inst{\ref{inst:louisiana}}  % confirmed
+  \and
+% Other contributors, alphabetical, including coordination meeting
+% participants
+K. Azalee Bostroem\inst{\ref{inst:stsci}}  % confirmed
+  \and
+Doug Burke\inst{\ref{inst:cfa}}  % confirmed
```
Comment (DougBurke) [inline astropy.tex:101]: Can you change me to 

 D. J. Burke

thanks?

File: astropy.tex (hunk 9 above)
Comment (astrofrog) [inline astropy.tex:410]: Done

File: astropy.tex (hunk 13 above)
Comment (astrofrog) [inline astropy.tex:531]: Done

**Comment by aptak** [comment]
I think overall the paper is very nice!  However I have some non-trivial comments, feel free to take with a huge grain of salt since they are a matter of opinion concerning presentation.  Also sorry if these issues had already been discussed.  Namely I see two audiences for this paper, and to some extent for astropy itself, 1) experienced astronomy python users and 2) astronomers using other languages, probably most often IDL.  I think the paper addresses 1) much more so than 2).  To address this I think the introduction should emphasize more so that the result of astropy + matplotlib + numpy is a fairly complete astronomical analysis system on par with, if not superior to, IDL + idlastro.  

Also, even for pythonistas I'm worried that starting with quantity as an example is too gimmicky. I personally was not that familiar with it yet and am impressed by the potential, but come away thinking that astropy is too verbose for typical use.  I.e., to what extent is this just a convenience for unit conversion that does not impact efficiency vs. a necessary aspect of using astropy?  I couldn't imagine the latter but again I am putting myself in the shoes of a newbie and fig 1 is the first code the reader is seeing.  So presumably arrays of quantities can be manipulated numerically, i.e., two arrays of distances can be added, either directly as python objects or at least you can easily extract an array of numpy floats to add.  Otherwise does what is shown in figure 1 scale to large tables of data, computing on pixel values in (fluxed) images and spectra, etc.?

Following on this point, as the intro to section 2.4 says, tables and arrays are the most common form of data that astronomers use so I think the paper should lead with that.  I think the first example should be a table with coordinates and fluxes or magnitudes for sources in a galaxy, maybe stating with the data can be easily read from a text file or fits file. Then you can show off being able to manipulate those quantities, say by assuming a distance or redshift (or showed that switching between the two is trivial) and adding a column with luminosities and linear distance offsets from the center of the galaxy to the table.  And then the table can be saved to FITS, VOTable, cvs, latex(?), etc.  This to me would be a very common type of operation that astronomers would do and would grab the attention of the reader.  Then the current examples going into more detail for quantities, coordinates, time, etc. could follow to show them off more.

And also following on this point, this may be the first the some readers have heard of pyfits / now astropy.io.fits.  Section 2.5.1 is almost entirely geared toward someone already using pyfits. 2.5.1. and the example in Figure 6 imply that it is easy to do common things like manipulate image data but I think this should be emphasized more clearly, e.g.,
hdus = fits.open("halpha.fits")
hdus_offband = fits.open("offband.fits")
onband = hdus[0].data
offband = hdus_offband[0].data
hdus[0].data = onband - offband
hdus.writeto("halpha_continuumsub.fits")
Maybe even add something like taking the median of both images and scaling offband by the ratio.
Or maybe showing off being able to read in an image and easily get the coordinates of a position, or give an example like that in 2.6.

Cheers,
Andy Ptak

File: astropy.tex (hunk 19 above)
Comment (astrofrog) [inline astropy.tex:508]: Done

**Comment by astrofrog** [comment]
Thanks to everyone for your comments! I can't reply to all of them, but I've addressed 99% of them, so if I didn't address something you brought up here, please let me know once I send the final draft out!
